### PR TITLE
Fixing problems reported by static analysis tool

### PR DESCRIPTION
**Pull request by jherico**
I ran a trial version of [PVS Studio](http://www.viva64.com/en/pvs-studio/) against our codebase and it discovered a number of suspicious items and some outright bugs.  

In a couple of places I was uncertain of the whether something represented an actual bug so I've added a FIXME to at least call out the suspicious code.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: d1a47ff75b68179639cf7459455641d3ddf56769 available at: https://gist.github.com/a3513cc31fe95ee58e32<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2bb1d1307aa246d04bdb7cd85d18587eab6fe5cc available at: https://gist.github.com/a3513cc31fe95ee58e32<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 31c37b5a3380111170f6567aa4fe696ee1a71955 available at: https://gist.github.com/a3513cc31fe95ee58e32<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 90fd0de8d0ebee5a0f09fa5ab98fc6e1c05cd32e available at: https://gist.github.com/a3513cc31fe95ee58e32<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e1afb86aa0ab80dffc239e4fc575d514cc687b72 available at: https://gist.github.com/a3513cc31fe95ee58e32<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.dmg

File: libraries/shared/src/GLMHelpers.cpp [hunk 1]
```diff
@@ -13,6 +13,24 @@
...
+const vec3 Vectors::UNIT_NEG_Y{ 0, -1, 0 };
+const vec3 Vectors::UNIT_NEG_Z{ 0, 0, -1 };
+const vec3 Vectors::UNIT_XY{ glm::normalize(UNIT_X + UNIT_Y) };
+const vec3 Vectors::UNIT_XZ{ glm::normalize(UNIT_X + UNIT_Z) };
+const vec3 Vectors::UNIT_YZ{ glm::normalize(UNIT_Y + UNIT_Z) };
+const vec3 Vectors::UNIT_XYZ{ glm::normalize(UNIT_X + UNIT_Y + UNIT_Z) };
+const vec3 Vectors::MAX{ FLT_MAX };
+const vec3 Vectors::MIN{ FLT_MIN };
+const vec3 Vectors::ZERO{ 0 };
+const vec3 Vectors::ONE{ 1 };
```
Comment (AndrewMeadows) [inline libraries/shared/src/GLMHelpers.cpp]: ZERO and ONE should use explicit floating point literals:  0.0f and 1.0f respectively.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 723cd35d53b1e857f443d5c4ac4961093a419f05 available at: https://gist.github.com/a3513cc31fe95ee58e32<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d311e4f2ea3a2f08cf9c05d60314125a9298ccd9 available at: https://gist.github.com/a3513cc31fe95ee58e32<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5645.dmg

**Pull request merged**